File: workspaces/clients/php/workspaces/laravel/tests/Browser/InteractsWithJavascriptTest.php
<?php

namespace Tests\Browser;

use PHPUnit\Framework\Assert;
use Puth\Laravel\Browser;
use Tests\Browser\Pages\Playground;
use Tests\PuthTestCase;

class InteractsWithJavascriptTest extends PuthTestCase
{
    public static bool $debug = false;

    function test_script()
    {
        $this->browse(function (Browser $browser) {
            $response = $browser->visit(new Playground)
                ->evaluate([
                    '1 + 1',
                    'window.document.location.href',
                ]);
            
            Assert::assertEquals([2, (new Playground)->url()], $response);
        });
    }
}

File: workspaces/clients/java/client/src/test/java/io/puth/InteractsWithJavascriptTest.java
package io.puth;

import org.junit.jupiter.api.Test;

import java.util.List;

import static org.junit.jupiter.api.Assertions.assertArrayEquals;

public class InteractsWithJavascriptTest extends BaseTest {

    @Test
    void test_script() {
        Object[] response = browser.visit(playground())
                .evaluate(new String[]{
                        "1 + 1",
                        "window.document.location.href"
                });

        assertArrayEquals(new Object[]{2, "https://playground.puth.dev/"}, response);
    }
}
